plugins/chain_plugin/chain_plugin.cpp

```diff
@@ -991,6 +996,25 @@ void chain_plugin::plugin_initialize(const variables_map& options) {
       if ( options.count("read-mode") ) {
          my->chain_config->read_mode = options.at("read-mode").as<db_read_mode>();
       }
+      my->p2p_accept_transactions = options.at( "p2p-accept-transactions" ).as<bool>();
+      my->api_accept_transactions = options.at( "api-accept-transactions" ).as<bool>();
+
+      if( my->chain_config->read_mode == db_read_mode::IRREVERSIBLE || my->chain_config->read_mode == db_read_mode::READ_ONLY ) {
+         if( my->chain_config->read_mode == db_read_mode::READ_ONLY ) {
+            wlog( "read-mode = read-only is deprecated use p2p-accept-transactions = false, api-accept-transactions = false instead." );
+         }
+         if( my->p2p_accept_transactions ) {
+            my->p2p_accept_transactions = false;
+            std::stringstream ss; ss << my->chain_config->read_mode;
+            wlog( "p2p-accept-transactions set to false due to read-mode: ${m}", ("m", ss.str()) );
+         }
+         if( my->api_accept_transactions ) {
+            my->api_accept_transactions = false;
+            std::stringstream ss; ss << my->chain_config->read_mode;
+            wlog( "api-accept-transactions set to false due to read-mode: ${m}", ("m", ss.str()) );
+         }
+         EOS_ASSERT( no_transactions(), plugin_config_exception, "IRREVERSIBLE should configure no_transactions" );
```

Comment:
This assertion could be moved to the beginning of `chain_plugin::plugin_startup` since by then all plugins had a chance to initialize and so the return value of `no_transactions()` should have finalized and therefore remain stable for the remainder of the process.

Reply:
Done.